apis/v1alpha1/gateway_types.go

```diff
@@ -117,6 +117,15 @@ type GatewaySpec struct {
 // defined protocols. If an implementation does not support a
 // specified protocol, it should raise a "ConditionUnsupportedProtocol"
 // condition for the affected Listener.
+//
+// Valid ProtocolType values are:
+//
+// * "HTTP"
+// * "HTTPS"
+// * "TLS"
+// * "TCP"
+//
+// +kubebuilder:validation:Enum=HTTP;HTTPS;TLS;TCP
```

Comment:
It looks like this is only affecting the spec docs, not actually resulting in changes to the CRD yaml. Does that mean that this will not actually going to result in this validation working? Maybe we need to update our CRD generation?

Reply:
It's a no-op because the validation was originally on the field, and the CRD generation still applies it there.